### PR TITLE
feat: refactor nip-04 screens

### LINKED ISSUE #2889: Improve NIP07 encrypt prompt
### Feature description

The current encrypt prompt only asks you to allow the client to encrypt things, but it doesn't show any details:

image

### Describe the solution

There are 2 things that could be added here: 
 - The message that is being encrypted
 - Which public key this will be encrypted for

### PR DESCRIPTION
**Pull request by pavanjoshi914**
### Describe the changes you have made in this PR

imrprove nip04 screens

### Link this PR to an issue [optional]

Fixes #2889 

### Type of change

(Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

image
image
image

### How has this been tested?

trying encrypt and decrypt on nostr chat

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/app/screens/Nostr/Confirm.tsx [hunk 1]
```diff
@@ -62,21 +72,29 @@ function NostrConfirm() {
...
+              />
+            )}
+            <div className="flex justify-center mb-4 gap-4">
+              <Hyperlink onClick={toggleShowDetails}>
+                {showDetails ? t("hide_details") : t("view_details")}
+              </Hyperlink>
             </div>
+            {showDetails && (
+              <div className="whitespace-pre-wrap break-words p-2 mb-4 shadow bg-white rounded-lg dark:bg-surface-02dp text-gray-500 dark:text-gray-400">
+                {tPermissions("nostr.peer")}: {peer}
```
Comment (bumi) [inline src/app/screens/Nostr/Confirm.tsx:95]: is there always a peer in this confirm screen?

Comment (pavanjoshi914) [reply src/app/screens/Nostr/Confirm.tsx:95]: yup! cause confirm screen is only used for encrypt and decrypt actions. for other nostr actions we have different screens

Comment (bumi) [reply src/app/screens/Nostr/Confirm.tsx:95]: should we then name the screen EncryptDecryptConfirm? or something?

Comment (pavanjoshi914) [reply src/app/screens/Nostr/Confirm.tsx:95]: resolved

File: src/extension/background-script/actions/nostr/decryptOrPrompt.ts [hunk 2]
```diff
@@ -33,9 +33,13 @@ const decryptOrPrompt = async (message: MessageDecryptGet, sender: Sender) => {
...
       }>({
         ...message,
-        action: "public/nostr/confirm",
+        action: "public/nostr/confirmEncryptOrDecrypt",
         args: {
-          description: i18n.t("permissions:nostr.nip04decrypt"),
+          description: {
+            action: i18n.t("permissions:nostr.nip04decrypt"),
+            peer: message.args.peer,
+            message: message.args.ciphertext,
```
Comment (bumi) [inline src/extension/background-script/actions/nostr/decryptOrPrompt.ts:41]: @reneaaron what do you think. 
This ciphertext will not be useful for the user. so I'd say we should leave this out here. What do you think?

Comment (reneaaron) [reply src/extension/background-script/actions/nostr/decryptOrPrompt.ts:40]: I agree, but then we'd need to have an alternative layout here. I'd suggest creating a follow-up task to split encrypt / decrypt into two different flows. I'd rather have them separate...

Comment (reneaaron) [reply src/extension/background-script/actions/nostr/decryptOrPrompt.ts:40]: Created an issue cc @pavanjoshi914 #2942

**Pull request merged**